src/Components/Components/src/Routing/LegacyRouteMatching/LegacyRouteTableFactory.cs

```diff
@@ -26,7 +27,7 @@ public static LegacyRouteTable Create(IEnumerable<Assembly> assemblies)
                 return resolvedComponents;
             }
 
-            var componentTypes = key.Assemblies.SelectMany(a => a.ExportedTypes.Where(t => typeof(IComponent).IsAssignableFrom(t)));
+            var componentTypes = RouteTableFactory.GetRouteableComponents(key.Assemblies);
```

Comment:
Is it possible to leave this file unchanged? The whole idea of the `Legacy*` variants of the routing logic is that they matched an exact point in time that we shipped before, and we keep them decoupled from the newer code so we don't have to reason about how any changes in the new code might impact behaviors in the legacy code.

Eventually (.NET 7?) we'll just remove the whole of the legacy version.

Reply:
👍🏽